mirascope/prompts.py

```diff
@@ -78,16 +93,12 @@ def template(cls) -> str:
 
     def __str__(self) -> str:
         """Returns the docstring prompt template formatted with template variables."""
-        template = self.template()
-        template_vars = [
-            var for _, var, _, _ in Formatter().parse(template) if var is not None
-        ]
-        return template.format(**{var: getattr(self, var) for var in template_vars})
+        return _format_template(self, self.template())
 
     @property
-    def messages(self) -> list[tuple[str, str]]:
+    def messages(self) -> list[ChatCompletionMessageParam]:
```

Comment:
if user passes in a list dict [str, Any] does mypy throw an error?

Reply:
Do you mean if they write their own `messages` property and try to return a base dict[str, Any]?

Reply:
yeah

Reply:
No because it's a `TypedDict` and not a pydantic model.